yt/frontends/amrvac/data_structures.py

```diff
@@ -247,81 +242,50 @@ def _set_code_unit_attributes(self):
-            kb = boltzmann_constant_cgs.value
-            mu0 = 4*np.pi
-
         # First check if overrides have been supplied, if that's the case use those instead.
-        # note: unit_length, unit_velocity and unit_temperature are passed to Dataset._override_code_units(), but
-        # unit_numberdensity is not. This is handled here, together with a similar mylog.info() message.
-        # Retrieve them from self.units_override as (val, unit) tuple to be consistent with YT.
-        if bool(self.units_override):
-            if self._usr_unit_system == 'mks':
-                unit_length = self.units_override.get('length_unit', (1, 'm'))
-                unit_numberdensity = self.units_override.get('numberdensity_unit', (1, 'm**-3'))
-                unit_velocity = self.units_override.get('velocity_unit', (0, 'm*s**-1'))
-            else:
-                unit_length = self.units_override.get('length_unit', (1, 'cm'))
-                unit_numberdensity = self.units_override.get('numberdensity_unit', (1, 'cm**-3'))
-                unit_velocity = self.units_override.get('velocity_unit', (0, 'cm*s**-1'))
+        # Assume cgs values and let YT handle conversion if supplied in an 'mks' unit system.
+        if self.units_override:
+            unit_length = self.units_override.get('length_unit', (1, 'cm'))
+            unit_numberdensity = self.units_override.get('numberdensity_unit', (1, 'cm**-3'))
+            unit_velocity = self.units_override.get('velocity_unit', (0, 'cm*s**-1'))
             unit_temperature = self.units_override.get('temperature_unit', (1, 'K'))
-
             mylog.info('Overriding numberdensity_unit: {:1.0e} {}.'.format(unit_numberdensity[0], unit_numberdensity[1]))
-            # use only values to calculate normalisations, units are handled afterwards
-            unit_length = unit_length[0]
-            unit_numberdensity = unit_numberdensity[0]
-            unit_velocity = unit_velocity[0]
-            unit_temperature = unit_temperature[0]
         else:
             # Next, check for unit normalisations in the .dat file. Do not use .get() with default value added
             # to explicitly print a warning to the console.
             try:
-                unit_length = self.parameters['unit_length']
-                unit_numberdensity = self.parameters['unit_numberdensity']
-                unit_temperature = self.parameters['unit_temperature']
-                unit_velocity = self.parameters['unit_velocity']
-            # If not present, use default values
+                unit_length = (self.parameters['unit_length'], 'cm')
+                unit_numberdensity = (self.parameters['unit_numberdensity'], 'cm**-3')
+                unit_temperature = (self.parameters['unit_temperature'], 'K')
+                unit_velocity = (self.parameters['unit_velocity'], 'cm*s**-1')
+            # If nothing present in datfile, use default values and print warning
             except KeyError:
```

Comment:
actually we know that those parameter are never present in the datfile (at least for now), so it's bad design to write this piece of code as if it was the default. More generally, we should not write more that the strictly necessary amount of code, so I'm strongly in disfavour of supporting hypothetical future features in amrvac.